src/components/views/rooms/RoomPreviewBar.js

```diff
@@ -89,15 +105,151 @@ module.exports = React.createClass({
+                secondaryActionLabel = _t("Sign In");
+                secondaryActionLabel = this.props.onSignInClick;
+                break;
+            }
+            case MessageCase.Kicked: {
+                const info = this._getKickOrBanInfo();
+                title = _t("You were kicked from this room by %(memberName)", info);
+                subTitle = _t("Reason: %(reason)", info);
+                primaryActionLabel = _t("Re-join");
+                primaryActionLabel = this.props.onJoinClick;
+                secondaryActionLabel = _t("Forget this room");
+                secondaryActionLabel = this.props.onForgetClick;
+                break;
+            }
+            case MessageCase.Banned: {
+                const info = this._getKickOrBanInfo();
+                title = _t("You were banned from this room by %(memberName)", info);
+                subTitle = _t("Reason: %(reason)", info);
+                primaryActionLabel = _t("Forget this room");
+                primaryActionLabel = this.props.onForgetClick;
+                break;
+            }
+            case MessageCase.OtherThreePIDError: {
+                break;
+            }
+            case MessageCase.MismatchThreePIDInvite: {
+                title = _t("The room invite wasn't sent to your account");
+                subTitle = _t("Sign in with a different account, ask for another invite, or add the e-mail address %(email) to this account.", {email: this.props.invitedEmail});
+                break;
+            }
+            case MessageCase.Invite: {
+                title = _t("%(memberName) invited you to this room", {memberName: this.props.inviterName});
+                primaryActionLabel = _t("Accept");
+                primaryActionLabel = this.props.onJoinClick;
+                secondaryActionLabel = _t("Reject");
+                secondaryActionLabel = this.props.onRejectClick;
+                break;
+            }
+            case MessageCase.ViewingRoom: {
+                if (this.props.canPreview) {
+                    title = _t("You are previewing this room. Want to join it?");
+                } else {
+                    title = _t("This room can't be previewed. Do you want to join it?");
+                }
+
+                title = _t("%(memberName) invited you to this room", {memberName: this.props.inviterName});
+                primaryActionLabel = _t("Accept");
+                primaryActionLabel = this.props.onJoinClick;
+                secondaryActionLabel = _t("Reject");
+                secondaryActionLabel = this.props.onRejectClick;
+                break;
+            }
+            case MessageCase.RoomNotFound: {
+                break;
+            }
+            case MessageCase.OtherError: {
+                break;
+            }
+        }
+
         let joinBlock; let previewBlock;
 
         if (this.props.spinner || this.state.busy) {
-            const Spinner = sdk.getComponent("elements.Spinner");
```

Comment:
Hmm, I am assuming the usage below is removed in the next commit...

Reply:
Only used on line 398 now, with a matching getComponent.